src/components/MealplanList.js

```diff
@@ -77,9 +77,13 @@ const mealReducer = (state, action) => {
   }
 }
 
-export default function MealplanList({ mealplanList }) {
+export default function MealplanList({ mealplanList, onChange }) {
   const [list, dispatch] = useReducer(mealReducer, mealplanList)
 
+  useEffect(() => {
+    onChange(list)
+  }, [list, onChange])
```

Comment:
Spannend das so zu sehen! Das könnte meine ganzen ungewollten re-renders beheben 😅 das onChange hier nochmal als dependency reinzuschreiben, macht/verhindert was genau?
Ab hier spekuliere ich ein wenig, bitte ignorieren wen ich falsch liege - Wenn das onChange(list) hier nur das handleChange in der App aufruft(?) und deren einzige function es ist ins localStorage zu schreiben, vielleicht kannst du das hier noch eindeutiger benennen? Nur da es sonst schon sehr viel onChange in dieser Datei gibt, die tatsächlich was "changen".

Reply:
Das onChange habe ich mit dazu gegeben, weil es, soweit ich das richtig verstanden habe, sich empfiehlt alles, was im Effect verwendet wird auch als dependency mitzugeben. Hat sonst keinen größeren Gedanken. Man packt das halt mit rein, damit das bei Änderungen auch mit abgedeckt ist. (Es ist aber schon relativ unwahrscheinlich, dass sich an dem onChange was ändert. Könnte es also auch weglassen) 

Die Namensgebungsproblematik kann ich total nachvollziehen. Hab schon überlegt es onMealplanListChange zu nennen... aber da das ja direkt in der MealplanList component ist, finde ich das auch nicht wirklich perfekt. Ich brainstorme da noch ein bisschen. Vielleicht komme ich ja noch auf einen eindeutigeren Namen. 😅 Danke für den Hinweis! 👍